### PR TITLE
bump xsimd to 8.0.3

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **xsimd/8.0.3**

Because xsimd/8.0.0 has some breaking changes, 
test_package.cpp have to be modified by version number.

I don't think versions of xsimd prior to 8.0.0 are compatible with macOS m1. 
(This is not official information).
When 7.5.0 was added to the xsimd recipe, no CI to macOS m1 was performed.

---
- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/xsimd/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -4,17 +4,26 @@ project(test_package CXX)
 include(CheckCXXCompilerFlag)
 
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup(TARGETS)
+conan_basic_setup()
```
[the remark this answers is not in the document]
Comment (uilianries) [reply recipes/xsimd/all/test_package/CMakeLists.txt:7]: keep `conan_basic_setup(TARGETS)`

[the remark this answers is not in the document]
Comment (toge) [reply recipes/xsimd/all/test_package/CMakeLists.txt:7]: sorry.
I misunderstood.
I will fix later.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

File: recipes/xsimd/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -9,12 +9,21 @@ conan_basic_setup(TARGETS)
 find_package(xsimd REQUIRED CONFIG)
 
 add_executable(${PROJECT_NAME} test_package.cpp)
-target_link_libraries(${PROJECT_NAME} xsimd)
+target_link_libraries(${PROJECT_NAME} xsimd::xsimd)
```
Comment (SpaceIm) [inline recipes/xsimd/all/test_package/CMakeLists.txt:12]: no, it's not the official target name.

**Pull request merged**